config/api.go

```diff
@@ -5,11 +5,32 @@ import (
 	"database/sql"
 	"database/sql/driver"
 	"encoding/json"
+	"fmt"
 	"net/url"
 	"strings"
 	"time"
 )
 
+type ConsumerType string
+
+const (
+	PullConsumer ConsumerType = "pull"
+	PushConsumer ConsumerType = "push"
+)
+
+func GetConsumerTypeFromString(value string) (ConsumerType, error) {
+	switch value {
+	case string(PullConsumer):
+		return PullConsumer, nil
+	case string(PushConsumer):
+		return PushConsumer, nil
+	case "":
+		return PushConsumer, nil
+	default:
+		return "", fmt.Errorf("invalid consumer type : %v", value)
+	}
+}
```

Comment:
I would consider using iota (integer) based enum implementation similar to [JobStatus](https://github.com/newscred/webhook-broker/blob/291cd1ea36165f9e5245383d0ae5614fee23f470/storage/data/job.go#L9).

Reply:
Bhaia, I have one concern about using iota based Enums. From my understanding, if a developer accidentally reshuffles the order of States [here](https://github.com/newscred/webhook-broker/blob/main/storage/data/job.go#L29-L35), it should mess up the entire existing database. Still the unit tests and integration tests won't catch that, because they all start with fresh data. Is my understanding correct? If so, isn't it a bit risky?

Reply:
Copying the discussion from Teams -
```
Yes that is a risk, but the risk, IMHO, is equivalent to some accidentally switching const values. In either case we should have knowledge that it should not be done.
It isn't full proof either way is my point
```